### DISCUSSION TITLE
Assigning UDP header checksum randomly possible?

### DISCUSSION THREAD

**Discussion by TomHsiung** [strongswan/strongswan#1771] [Ideas]
Yesterday I installed Cloudflare Warp clients and I captured its packet flow. The encrypted packets do have UDP headers. I noticed a difference between it and strongswan.

For strongswan, the value of the checksum section of the UDP header is always 0x0000. Instead, UDP headers of Warp have various values of the checksum section, which is different from packet to packet. I don't know the purpose of this from Warp. Maybe it just is assigned randomly.

If so, is it possible for strongswan to do that? For security reasons, all encrypted traffics with a never changing UDP header checksum of 0x0000 is easily recognized by censors.

**Comment by tobiasbrunner** [comment]
Since strongSwan generally doesn't handle IPsec traffic, you might want to ask on the [Netdev mailing list](http://vger.kernel.org/vger-lists.html#netdev) about this.

> For security reasons, all encrypted traffics with a never changing UDP header checksum of 0x0000 is easily recognized by censors.

I think detecting UDP-encapsulated ESP isn't that hard either way.

**Reply by TomHsiung** [reply]
Oh. That is too bad for Chinese IPsec users.
